### PR TITLE
[opencv] enabled OpenVINO support

### LINKED ISSUE #28357: [opencv] OpenVino feature
****

**Is your feature request related to a problem? Please describe.**
I think there is no "feature" to build opencv with Intel's Inference Engine support (OpenVINO).

**Proposed solution**

I dont know if it easy to implement or if is posible but i would like the feature just will enable the WITH_OPENVINO option in opencv build comand:

I guess the portfile.cmake would be something like this in the vcpkg_check_features:

vcpkg_check_features(OUT_FEATURE_OPTIONS FEATURE_OPTIONS
 FEATURES
 "ade"       WITH_ADE
...
"openvino" WITH_OPENVINO 
...
 "dc1394"    WITH_1394
)

Thank you.

### PR DESCRIPTION
**Pull request by ilya-lavrenov**
**Reason:** Optimized OpenCV DNN module with OpenVINO, because by default OpenCV DNN module is slow as does not have any integrated performant backend.

- [ ] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [ ] SHA512s are updated for each updated download
- [ ] The "supports" clause reflects platforms that may be fixed by this new version
- [ ] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [ ] Any patches that are no longer applied are deleted from the port's directory.
- [ ] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [ ] Only one version is added to each modified port's versions file.

### CODE REVIEW

File: ports/opencv4/vcpkg.json [hunk 1]
```diff
@@ -109,6 +109,24 @@
...
+              "platform": "!(windows & arm)"
+            },
+            {
+              "name": "gpu",
+              "platform": "x64 & !(osx | uwp) & !static"
+            },
+            "hetero",
+            "ir"
+          ]
+        },
```
Comment (ilya-lavrenov) [inline ports/opencv4/vcpkg.json:129]: Hi @dan-shaw 

Could you please advice why openvino is still used with all features, while I select specific ones?
```
% ./vcpkg install opencv 
Computing installation plan...
The following packages will be built and installed:
  * liblzma:arm64-osx-dynamic -> 5.4.3#1
  * libpng:arm64-osx-dynamic -> 1.6.39#1
  * libwebp[core,libwebpmux,nearlossless,simd]:arm64-osx-dynamic -> 1.3.2
    opencv[core,default-features]:arm64-osx-dynamic -> 4.8.0
  * opencv4[core,default-features,dnn,jpeg,png,quirc,tiff,webp]:arm64-osx-dynamic -> 4.8.0#7
  * openvino[auto,auto-batch,core,cpu,hetero,ir,onnx,paddle,pytorch,tensorflow,tensorflow-lite]:arm64-osx-dynamic -> 2023.1.0
  * quirc:arm64-osx-dynamic -> 1.1#3
  * tiff[core,jpeg,lzma,zip]:arm64-osx-dynamic -> 4.5.1#2
  * zlib:arm64-osx-dynamic -> 1.3
Additional packages (*) will be modified to complete this operation.
Detecting compiler hash for triplet arm64-osx-dynamic...
```

File: ports/opencv4/0020-static-openvino.patch [hunk 2]
```diff
@@ -0,0 +1,25 @@
+diff --git a/cmake/OpenCVUtils.cmake b/cmake/OpenCVUtils.cmake
```
Comment (ilya-lavrenov) [inline ports/opencv4/0020-static-openvino.patch:1]: https://github.com/opencv/opencv/commit/ef9d14f181ad8cca71443beaf3874de3197d4e47

File: ports/opencv4/vcpkg.json (hunk 1 above)
Comment (ilya-lavrenov) [reply ports/opencv4/vcpkg.json:129]: I'm using the following syntax https://learn.microsoft.com/en-us/vcpkg/reference/vcpkg-json#dependency-features

Comment (dg0yt) [reply ports/opencv4/vcpkg.json:129]: > Could you please advice why openvino is still used with *all* features, while I select specific ones?

You probably want to say "with *default* features".

This is a port manifest, not a user project manifest. 
In a port manifest, `"default-features": "false"` doesn't immediately deselect any features. It merely declares that you explicitly listed all features which are forming this dependency. You put control in the hand of the user. :+1:

If the user doesn't care (or if it is vcpkg CI), the user gets the default features of openvino. (And that's the meaning of default here.)
In classic mode, the user can list `openvino[core]` when installing `opencv` to get rid of the default features.
In manifest mode, the user can list `openvino` with `"default-features": false` in the manifest.

Disclaimer: I'm only explaining the observed behaviour. It is a controversial topic which I would call: the three meanings of "default-features" 😕

**Comment by ilya-lavrenov** [comment]
Hi @dg0yt 
Could you please advice what should I do with?
```
##[error]REGRESSIONS:
REGRESSION: vcpkg-ci-opencv:x64-uwp cascaded, but it is required to pass. (C:\a\1\s\scripts\azure-pipelines/../ci.baseline.txt).
```

**Comment by dg0yt** [comment]
> Hi @dg0yt Could you please advice what should I do with?
> 
> ```
> ##[error]REGRESSIONS:
> REGRESSION: vcpkg-ci-opencv:x64-uwp cascaded, but it is required to pass. (C:\a\1\s\scripts\azure-pipelines/../ci.baseline.txt).
> ```

You must resolve it ;-)
vcpkg-ci-opencv ensures testing of non-default features of opencv. The addition of openvino seems to have added some dependency which is not supported on uwp, so the required build can't even start. This is a regression. You will probably have to add more control with features or `platform`.

You can access the test ports as overlay ports. You can inspect `depend-info` or `install --dry-run` as needed to find out more, even if you don't have a uwp toolchain.

**Pull request closed without merging**